docs/tutorial/units.rst

```diff
@@ -4,82 +4,133 @@
+
+Adding Units to Data
+====================
+
 To use units we must first import them from Astropy.
-To save on typing it's standard practice to import the units module as ``u``::
+Note that it is standard practice to import the units module as ``u``::
 
    >>> import astropy.units as u
 
-Once we have imported the units module we can create a `~astropy.units.Quantity` by multiplying a number by a unit::
+We can create a `~astropy.units.Quantity` by multiplying a number by a unit::
 
    >>> length = 10 * u.meter
    >>> length
    <Quantity 10. m>
 
-A `~astropy.units.Quantity` has both a ``.unit`` and a ``.value`` attribute::
+A `~astropy.units.Quantity` can be decomposed into its unit and numerical value using the ``.unit`` and ``.value`` attributes, respectively::
 
   >>> length.value
   10.0
 
   >>> length.unit
   Unit("m")
 
-These `~astropy.units.Quantity` objects can also be converted to other units or unit systems::
-
-  >>> length.to(u.km)
-  <Quantity 0.01 km>
-
-  >>> length.cgs
-  <Quantity 1000. cm>
+Arithmetic Operations With Units
+================================
 
-Probably most usefully, `~astropy.units.Quantity` objects will propagate units through arithmetic operations when appropriate::
+`~astropy.units.Quantity` objects propagate units through arithmetic operations::
 
   >>> distance_start = 10 * u.mm
   >>> distance_end = 23 * u.km
-  >>> length = distance_end - distance_start
-  >>> length
+  >>> displacement = distance_end - distance_start
+  >>> displacement
   <Quantity 22.99999 km>
 
   >>> time = 15 * u.minute
-  >>> speed = length / time
+  >>> speed = displacement / time
   >>> speed
   <Quantity 1.53333267 km / min>
 
-However, operations which do not make physical sense for the units specified will cause an error::
+However, operations with incompatible units raise an error::
 
-  >>> length + time
+  >>> displacement + time
   Traceback (most recent call last):
   ...
   astropy.units.core.UnitConversionError: Can only apply 'add' function to quantities with compatible dimensions
 
+Converting Units
+================
+
+These `~astropy.units.Quantity` objects can also be converted to other units or unit systems::
+
+  >>> length.to(u.km)
+  <Quantity 0.01 km>
+
+  >>> length.cgs
+  <Quantity 1000. cm>
+
+Unit Equivalencies
+==================
+
+It is commonplace to convert between units which are only compatible under certain assumptions.
+For example, in spectroscopy, spectral energy and wavelength are equivalent given the relation
+:math:`E=hc/\lambda`.
+If we try to convert a wavelength to energy using what we learned in the previous section,
+we get an exception because length and energy are, in general, not compatible units,::
```

Comment:
```suggestion
If we try to convert a wavelength to energy using what we learned in the previous section, we get an exception because length and energy are, in general, not compatible units,::
```